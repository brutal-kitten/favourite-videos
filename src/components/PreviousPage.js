import React, { Component } from 'react';


class PreviousPage extends Component {

  //if there is previous page calls function that recalculate and set state (startIndex, currentPage) in Pagination component
  //if there isn't previous page user stays at the same page
  handleChange = (event) => {

    event.preventDefault();
    console.log("previousPage")
    let newPage = parseInt(this.props.currentPage, 10) - 1;
    if (newPage > 0) {
      this.props.setStartIndex(newPage);
    }
  }

  render () {

    return (
      <div className="previousPage">
        <button tabindex='0' className="buttonPreviousPage" onClick={(event) => this.handleChange(event)}>
          <span className="glyphicon glyphicon-chevron-left"></span>
        </button>
      </div>
    )
  }
}

export default PreviousPage
