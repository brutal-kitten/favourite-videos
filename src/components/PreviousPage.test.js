import React from 'react';
import ReactDOM from 'react-dom';
import PreviousPage from './PreviousPage.js';
import TestRenderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

it('renders properly', () => {
  const tree = TestRenderer.create(
    <PreviousPage />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should call setStartIndex with value of previous page if there is previous page(previous page can not be less then 1)', () => {
  const MockSetStartIndex = jest.fn();
  const props = {
    setStartIndex: MockSetStartIndex,
    currentPage: 2
  }

  const wrapper = shallow(<PreviousPage {...props}/>);

  wrapper.find('button').simulate('click', {
    preventDefault: () => {},
  });

  expect(MockSetStartIndex.mock.calls.length).toBe(1);
  expect(MockSetStartIndex.mock.calls[0][0]).toBe(1);
});

it('should not call setStartIndex if there is no previous page(previous page can not be less then 1)', () => {
  const MockSetStartIndex2 = jest.fn();
  const props2 = {
    setStartIndex: MockSetStartIndex2,
    currentPage: 1
  }

  const wrapper = shallow(<PreviousPage {...props2}/>);

  wrapper.find('button').simulate('click', {
    preventDefault: () => {},
  });

  expect(MockSetStartIndex2.mock.calls.length).toBe(0);
});
