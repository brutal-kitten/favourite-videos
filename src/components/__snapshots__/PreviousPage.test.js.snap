// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<div
  className="previousPage"
>
  <button
    className="buttonPreviousPage"
    onClick={[Function]}
    tabIndex="0"
  >
    <span
      className="glyphicon glyphicon-chevron-left"
    />
  </button>
</div>
`;
